compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1403,9 +1404,7 @@ pub enum CastKind {
     /// * [`PointerCoercion::MutToConstPointer`]
     ///
     /// Both are runtime nops, so should be [`CastKind::PtrToPtr`] instead in runtime MIR.
-    PointerCoercion(PointerCoercion),
-    /// Cast into a dyn* object.
-    DynStar,
+    PointerCoercion(PointerCoercion, CoercionSource),
```

Comment:
I'm not well versed in `dyn*` object, does grouping them with pointers actually make sense? cc @compiler-errors